### ISSUE TITLE
OAuth2 request doesn't support special characters in password

### ISSUE THREAD

**Issue by ddBearUA** [gitlab4j/gitlab4j-api#343]
Passing a password with special characters (e.g. ~~german umlauts~~, backslash or double quotes) via the OAuth2 login causes an response with the status Unauthorized (HTTP Status-Code 401).

**Comment by gmessner** [comment]
@ddBearUA 
I have been unable to get this to fail with the special chars I have tested with.  Could you provide me a phrase with the characters you have problems with? 

I will test with those characters and fix any issues.

**Comment by gmessner** [comment]
@ddBearUA 
I've done further testing using "ausgefüllt" as the password.  What I did encounter was a problem reading the password from the properties file, Once I corrected that, all worked.  Can you make sure that the password is being read correctly from your configuration file?

I had to change:
```
InputStream input = new FileInputStream(propertiesFile);
// Resulted in "ausgefÃ¼llt" (WRONG)
```
to:
```
InputStreamReader input = new InputStreamReader(new FileInputStream(propertiesFile));
// Resulted in "ausgefüllt" (CORRECT)
```

**Comment by ddBearUA** [comment]
@gmessner 
Sorry for the confusion. The problem was not an umlaut but a backslash in the password.

You can test it with a password "ab\tcd42". To exclude a problem reading password from a file I defined it directly in a char array. Following code leads to aan 401 response:
```
SecretString password = new SecretString(new char[] {'a', 'b', '\\', 't', 'c', 'd', '4', '2'});
GitLabApi.oauth2Login(host, user, password);
```
As a workaround the backslash can be escaped. The following code performs a login for the user with password "ab\tcd42":
```
SecretString password = new SecretString(new char[] {'a', 'b', '\\', '\\', 't', 'c', 'd', '4', '2'});
GitLabApi.oauth2Login(host, user, password);
```

P.S. A double quote character `"` in a password causes a bed request while OAuth login.

**Comment by gmessner** [comment]
@ddBearUA 
Special chars are now properly handled with OAUTH2 logins.  Unit tests have also been added to ensure that passwords with special chars are properly encoded when sent to the GitLab server.  This fix is in Release 4.10.12 that was just pushed to maven central.

When you have a chance, please verify and close this issue.

**Comment by ddBearUA** [comment]
Verified in release 4.10.14 with password that contains backslash and double quote.
